sr_module.c

```diff
@@ -199,28 +199,48 @@ int register_module(const struct module_exports* e, char* path, void* handle)
 
 static inline int version_control(const struct module_exports* exp, char *path)
 {
-	if ( !exp->version ) {
+	const char *hint = "(try `make clean all modules' and reinstall everything)";
+	if ( !exp->ver_info.version ) {
 		LM_CRIT("BUG - version not defined in module <%s>\n", path );
 		return 0;
 	}
-	if ( !exp->compile_flags ) {
+	if ( !exp->ver_info.compile_flags ) {
 		LM_CRIT("BUG - compile flags not defined in module <%s>\n", path );
 		return 0;
 	}
+	if ( !exp->ver_info.scm.type ) {
+		LM_CRIT("BUG - SCM type not defined in module <%s> %s\n", path, hint );
+		return 0;
+	}
+	if ( !exp->ver_info.scm.rev ) {
+		LM_CRIT("BUG - SCM revision not defined in module <%s> %s\n", path, hint );
+		return 0;
+	}
 
-	if (strcmp(OPENSIPS_FULL_VERSION, exp->version)==0){
-		if (strcmp(OPENSIPS_COMPILE_FLAGS, exp->compile_flags)==0)
-			return 1;
-		else {
-			LM_ERR("module compile flags mismatch for %s "
-				" \ncore: %s \nmodule: %s\n",
-				exp->name, OPENSIPS_COMPILE_FLAGS, exp->compile_flags);
-			return 0;
-		}
+	if (strcmp(OPENSIPS_FULL_VERSION, exp->ver_info.version)!=0) {
+		LM_ERR("module version mismatch for %s; core: %s; module: %s\n",
+			exp->name, OPENSIPS_FULL_VERSION, exp->ver_info.version );
+		return 0;
 	}
-	LM_ERR("module version mismatch for %s; core: %s; module: %s\n",
-		exp->name, OPENSIPS_FULL_VERSION, exp->version );
-	return 0;
+	if (strcmp(OPENSIPS_COMPILE_FLAGS, exp->ver_info.compile_flags)!=0) {
+		LM_ERR("module compile flags mismatch for %s "
+			" \ncore: %s \nmodule: %s\n",
+			exp->name, OPENSIPS_COMPILE_FLAGS, exp->ver_info.compile_flags);
+		return 0;
+	}
+	if (strcmp(core_scm_ver.type, exp->ver_info.scm.type) != 0) {
+		LM_ERR("module SCM type mismatch for %s "
```

Comment:
The term of `SCM` may be confusing to regular user (even for me, it was unknown by now). The idea of a log is to be clear and understandable by the user. So, let's try to explain / replace this SCM with something that can be understood by user - even like GIT revision, version, etc...something to ring a  :bell:

Reply:
Thanks for reminding how old I am, Bogdan! 😂 Will do.

Reply:
Adjusted as suggested. Thanks, @bogdan-iancu!